### PR TITLE
Stax

### PR DESCRIPTION
**Pull request by sgliner-ledger**
Stax porting and ragger tests ! :rocket:

### CODE REVIEW

File: src/get_address_nbgl.c [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        approve_callback();
+    }
+    else {
+        cancel_callback();
+    }
+}
+
+static void gen_address(void)
+{
+    if (!_generated) {
```
Comment (xchapron-ledger) [inline src/get_address_nbgl.c:52]: I don't get the point of this? Even on Bagl this doesn't seem's to be necessary?

File: src/menu_nbgl.c [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+static const char* const infoTypes[] = {"Version", "Developer", "Copyright"};
+static const char* const infoContents[] = {APPVERSION, "Jspada", "(c) 2022 Ledger"};
+
+static bool navigation_cb(uint8_t page, nbgl_pageContent_t* content) {
+    UNUSED(page);
+    content->type = INFOS_LIST;
+    content->infosList.nbInfos = 3;
+    content->infosList.infoTypes = (const char**) infoTypes;
+    content->infosList.infoContents = (const char**) infoContents;
```
Comment (xchapron-ledger) [inline src/menu_nbgl.c:14]:
```suggestion
    content->infosList.infoTypes = infoTypes;
    content->infosList.infoContents = infoContents;
```

File: src/menu_nbgl.c [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+void ui_menu_about(void) {
+    nbgl_useCaseSettings("Mina", 0, 1, false, ui_idle, navigation_cb, NULL);
+}
+
+void ui_idle(void) {
+#ifdef HAVE_ON_DEVICE_UNIT_TESTS
+    nbgl_useCaseHome("Mina unit tests", &C_Mina_64px, NULL, false, ui_menu_about, exit);
+#else // HAVE_ON_DEVICE_UNIT_TESTS
+    nbgl_useCaseHome("Mina", &C_Mina_64px, NULL, false, ui_menu_about, exit);
```
Comment (xchapron-ledger) [inline src/menu_nbgl.c:30]:
```suggestion
    nbgl_useCaseHome(APPNAME, &C_Mina_64px, NULL, false, ui_menu_about, exit);
```

File: tests/test_crypto.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env python3
+
+import pytest
+
+
+def test_crypto(backend):
+    apdu = bytearray.fromhex("e004000000")
+    assert(not backend.send_raw(apdu))
```
Comment (xchapron-ledger) [inline tests/test_crypto.py:8]: Can you explain what is this doing?

File: tests/test_fuzz.py [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+sys.path.append(os.path.dirname(os.path.realpath(__file__)) + "/../utils/")
+import mina_ledger_wallet as mina
+
+TESTS_ROOT_DIR = Path(__file__).parent
+
+
+def test_fuzz(test_name, backend, firmware, navigator):
+        # Invalid message 1
+        apdu = bytearray.fromhex("")
+        assert(not backend.send_raw(apdu))
```
Comment (xchapron-ledger) [inline tests/test_fuzz.py:22]: you are expecting a raise here? Or a false?

File: src/sign_tx_nbgl.c [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+    nbgl_useCaseStatus("TRANSACTION\nSIGNED", true, ui_idle);
+}
+
+static void cancel_callback(void)
+{
+    sendResponse(0, false);
+    nbgl_useCaseStatus("Transaction\ncancelled", false, ui_idle);
+}
+
+static void sign_transaction(void)
```
Comment (xchapron-ledger) [inline src/sign_tx_nbgl.c:41]: Same as for public key, could this be kept in sign_tx.c to be common?

File: src/utils.c [hunk 7]
```diff
@@ -72,90 +72,86 @@ int b58_encode(const unsigned char *in, unsigned char length,
...
-	for (i = 0; i < binsz; ++i)	{
-		if (binu[i])
-			break;
-		--*binszp;
-	}
-	*binszp += zerocount;
-
-	return true;
+
+int b58_decode(const char *in, size_t in_len, uint8_t *out, size_t out_len) {
```
Comment (xchapron-ledger) [inline src/utils.c:95]: might used the version from crypto_helpers?

Comment (sgliner-ledger) [reply src/utils.c:95]: I used the version from the standard app indeed

File: Makefile [hunk 8]
```diff
@@ -153,21 +111,30 @@ DEFINES       += HAVE_WEBUSB WEBUSB_URL_SIZE_B=$(shell echo -n $(WEBUSB_URL) | w
...
-DEFINES       += BUILD_NANOX IO_SEPROXYHAL_BUFFER_SIZE_B=300
-DEFINES       += HAVE_GLO096
-DEFINES       += HAVE_BAGL BAGL_WIDTH=128 BAGL_HEIGHT=64
-DEFINES       += HAVE_BAGL_ELLIPSIS # long label truncation feature
-DEFINES       += HAVE_BAGL_FONT_OPEN_SANS_REGULAR_11PX
-DEFINES       += HAVE_BAGL_FONT_OPEN_SANS_EXTRABOLD_11PX
-DEFINES       += HAVE_BAGL_FONT_OPEN_SANS_LIGHT_16PX
+    DEFINES += HAVE_BAGL HAVE_UX_FLOW
+    ifneq ($(TARGET_NAME),TARGET_NANOS)
+        DEFINES += BUILD_NANOX
```
[the remark this answers is not in the document]
Comment (sgliner-ledger) [reply Makefile:130]: Its used for crypto tests

File: src/main.c [hunk 9]
```diff
@@ -308,7 +317,10 @@ __attribute__((section(".boot"))) int main(void) {
     os_boot();
 
     for (;;) {
-        UX_INIT();
+                UX_INIT();
+#ifdef HAVE_BAGL
+                ux_stack_push();
```
[the remark this answers is not in the document]
Comment (sgliner-ledger) [reply src/main.c:322]: It's legacy code, I'll remove it :ok_hand:

**Pull request merged**